### ISSUE TITLE
Improve the architecture of the repository

### ISSUE THREAD

**Issue by Nelrohd** [ClearGDPR/ClearGDPR#375]
**Is your feature request related to a problem? Please describe.**
It's confusing to have demo folders and libs folder at the same level. Also, the demo contains some libs.

**Describe the solution you'd like**
Make a clear separation between demo folder and libs folder. Extract the libs from the demo.

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by Nelrohd** [comment]
@sindelio @fjcero Opinions please?

**Comment by Nelrohd** [comment]
Suggestion: https://github.com/ClearGDPR/ClearGDPR/wiki/Structure-of-the-repository

**Comment by sindelio** [comment]
I like the structure proposed!
My suggestions: 
1. tools -> infrastructure
2. lib -> library (might seem simple but not everyone know lib is short for library)
3. travis inside infrastructure
Also, each of these files should have nice readmes

**Comment by segoddnja** [comment]
I agree with Sindelio except one thing - I would use `packages` instead of `lib`, _cg_ and _admin_ are really not libraries.

**Comment by fjcero** [comment]
`packages` is kind of de facto standard for monorepo and projects with internal dependencies. I agree with the demo structure, perhaps changing `server` and `client` as labels. Makes sense to have general tooling inside the infrastructure layer, even travis. Each package will have specific configuration as well, so tooling could be in the same place
